drivers/overlay/overlay.go

```diff
@@ -162,3 +150,69 @@ func (d *driver) configure() error {
 func (d *driver) Type() string {
 	return networkType
 }
+
+func (d *driver) nodeJoin(node net.IP, self bool) {
+	d.Lock()
+	if self {
+		d.bindAddress = node.String()
+	} else {
+		d.neighIP = node.String()
+	}
+
+	networks := d.networks
+	d.Unlock()
+	if d.bindAddress != "" && !d.isSerfAlive() {
+		err := d.serfInit()
+		if err != nil {
+			logrus.Errorf("initializing serf instance failed: %v", err)
+			return
+		}
+
+		for _, n := range networks {
+			if n.sandbox() == nil {
+				continue
+			}
+			err = n.initVxlan()
+			if err != nil {
+				logrus.Errorf("initializing vxlan failed for %s: %v", n.id, err)
+			}
+		}
+	}
+
+	if d.serfInstance != nil && d.neighIP != "" {
+		err := d.serfJoin()
+		if err != nil {
+			logrus.Errorf("joining serf neighbor failed: %v", err)
+			return
+		}
+	}
+}
+
+func (d *driver) nodeLeave(host net.IP) {
```

Comment:
Let's remove all the dead code. If `nodeLeave` doesn't do anything then let's remove it